include/settings.hpp

```diff
@@ -0,0 +1,428 @@
+#ifndef GRAPH_PROTOTYPE_SETTINGS_HPP
+#define GRAPH_PROTOTYPE_SETTINGS_HPP
+
+#include <atomic>
+#include <chrono>
+#include <concepts>
+#include <mutex>
+#include <optional>
+#include <reflection.hpp>
+#include <set>
+#include <tag.hpp>
+#include <variant>
+
+namespace fair::graph {
+
+struct SettingsCtx {
+    // using TimePoint = std::chrono::time_point<std::chrono::utc_clock>; // TODO: change once the C++20 support is ubiquitous
+    using TimePoint               = std::chrono::time_point<std::chrono::system_clock>;
+    std::optional<TimePoint> time = std::nullopt; /// UTC time-stamp from which the setting is valid
+    std::string              context;             /// user-defined multiplexing context for which the setting is valid
+};
+
+template<typename T, typename Node>
+concept Setting = requires(T t, Node& n, std::span<const std::string> parameter_keys, const std::string &parameter_key, const tag_t::map_type &parameters, SettingsCtx ctx) {
```

Comment:
```suggestion
concept Settings = requires(T t, Node& n, std::span<const std::string> parameter_keys, const std::string &parameter_key, const tag_t::map_type &parameters, SettingsCtx ctx) {
```
Small naming nit: plural to reflect that this is/manages a set of settings and not just a single one?